core/domain/feedback_services.py

```diff
@@ -172,3 +173,94 @@ def get_thread_analytics(exploration_id):
     """
     return feedback_jobs.FeedbackAnalyticsAggregator.get_thread_analytics(
         exploration_id)
+
+def create_suggestion(exploration_id, author_id, exploration_version,
+                      state_name, suggestion_content):
+    """Creates a new SuggestionModel object and the corresponding
+    FeedbackThreadModel object."""
+
+    thread_id = feedback_models.FeedbackThreadModel.generate_new_thread_id(
+        exploration_id)
+    (feedback_models.SuggestionModel
+        .create(exploration_id, thread_id, author_id, exploration_version,
+                state_name, suggestion_content))
+    thread = feedback_models.FeedbackThreadModel.create(
+        exploration_id, thread_id)
+    thread.has_suggestion = True
+    thread.exploration_id = exploration_id
+    thread.state_name = state_name
+    thread.original_author_id = author_id
+    thread.status = feedback_models.STATUS_CHOICES_OPEN
+    thread.subject = "SUGGESTION"
+    thread.put()
+
+
+def get_suggestion(exploration_id, thread_id):
+    return feedback_models.SuggestionModel.get(
+        _get_instance_id(thread_id, exploration_id))
```

Comment:
At the domain level you'd just want something like

    `feedback_models.SuggestionModel.get_by_exploration_and_thread_id(exploration_id, thread_id)`

(I put `exploration_id` first since it's the larger 'outer' context.)